test/regression/targets/sc-15387.cc

```diff
@@ -0,0 +1,132 @@
+#include <tiledb/tiledb>
```

Comment:
Out of curiosity, why are we creating this new regression suite rather than adding this in the existing test suite?

Reply:
The current `tiledb_unit` is an agglomeration of many different things, including some regression tests. The idea of a specific regression test suite is that each one is added standalone, as "expected to fail" (can be marked as such in catch2), and then converted to "expected to pass" in some separate commit with the fix (they can be asynchronous and done by separate people - in the future anyway). As we refactor into component libraries with better isolation, ideally we will also look for and add any missing coverage in the unit tests which could have isolated and prevented the defect -- but those are a different category of test.

Two other goals:
- regression tests should only use the public API; `tiledb_unit` currently links everything
- separating into different test executables allows some parallelization of the test run (at least using ctest - may require some cmake changes)